textBasedAdventure.js

```diff
@@ -1,5 +1,583 @@
+  // console.log(wantBreakfast)
+}
+      
+  console.log("Let see what you have first as options.");
+
+let willUserChoose = readline.question("would you like to choose for your[self] or, be given a [random] option? ");
+//  for (let i = 0; i < recipes.length; i++) {
+  function getRecipe() {
+      
+    let recipes = ["oatmeal", "sandwiches", "smoothie bowls"];
+    let selectedRecipe = recipes[Math.floor(Math.random() * recipes.length)];
+    return selectedRecipe;
+
+  }
+  
+  if (willUserChoose === "random") {
+  
+      let selected = getRecipe();
+      console.log(selected)
+      choiceRunner(selected) 
+
+  } else if (willUserChoose === "self") {
+
+    let usersChoice = readline.question("OK! would you like [oatmeal], [sandwiches], or [smoothie] bowls? ");
+    choiceRunner(usersChoice);
+      
+  } else {
+    wantBreakfast = readline.question('Would you like to prepare a nice BREAKFAST to start your day, [y]es or [n]o? ');
+    // getAllRecipe();
+
+  }
+
+
+  askForGroceries = readline.question("Do you have all the groceries [y]es or [n]o? ");
+  if (askForGroceries === "y") {
+    console.log("Great let get started");
+    getRecipe();
+  } else {
+    console.log("Go do your groceries then come back :( ")
+  }
+  willUserChoose = readline.question("would you like to choose for your[self] or, be given a [random] option again? ");
+  usersChoice = readline.question("OK! would you like [oatmeal], [sandwiches], or [smoothie] bowls? ");
+  choiceRunner(usersChoice);
+
+  let willUserPrint = readline.question("would you like to print all the recipes [y]es or [n]o? ");
+  function getAllRecipe() {
+    console.clear();
+    ingredientsOvernightOat();
+    smoothieBowls();
+    fruitNutOatmeal();
+    eggBreakfastSandwich();
+    SmokedSalmonDillEggsSandwich();
+    GuacamoleBreakfastSandwich();
+  }
+  console.log(willUserPrint)
+  
+  if (willUserPrint === "y") {
+    
+    getAllRecipe()
+    
+  } else {
+
+    willUserChoose = readline.question("would you like to choose for your[self] or, be given a [random] option again? ");
+    usersChoice = readline.question("OK! would you like [oatmeal], [sandwiches], or [smoothie] bowls? ");
+    choiceRunner(usersChoice);
+
+  }
+
+
+
+  usersPantryCheatCheat = readline.question("OK! would you like pantry cheat cheat how to double recipes [y] or [n]? ");
+  console.log("Pantry Cheat Cheat How to Double Recipes")
+
+if (usersPantryCheatCheat === "y"){
+  const myForEach = ( arr, fn) => {
+    for ( let i = 0 ; i< arr.length ; i ++) {
+      fn([arr[i]])
+    }
+  }
+
+  const doubleRecipeTsp = (num) => {
+    console.log((num * 2)+ " tsp")
+  }
+
+  const doubleRecipeTbs = (num) => {
+    console.log((num * 2)+ " Tbs")
+  }
+
+  const doubleRecipeCups = (num) => {
+    console.log( (num * 2)+ " Cups")
+  }
+
+  console.log(`cheat cheat for teaspoon\n`)
+  myForEach([1 ,2, 3, 4], doubleRecipeTsp);
```

Comment:
Since you're using the array [1,2,3,4] twice you could of created a variable and just called it in both these instances.